ai_genomics/utils/entities.py

```diff
@@ -0,0 +1,50 @@
+"""Utilities for working with entities."""
+from collections import Counter
+from itertools import chain
+import numpy as np
+
+from typing import Mapping, Optional, Union
+
+
+def strip_scores(entities):
+    return {k: [e[0] for e in v] for k, v in entities.items()}
+
+
+def filter_entities(
+    entities: Mapping[str, Mapping[str, Union[str, str]]],
+    min_entity_freq: Optional[Union[int, float]] = None,
+    max_entity_freq: Optional[Union[int, float]] = None,
+) -> Mapping[str, Mapping[str, Union[str, str]]]:
+    """_summary_
+
+    Args:
+        entities (Mapping[str, Mapping[str, Union[str, str]]]): DBpedia entities
+            for a set of documents without scores.
+        min_entity_freq (Optional[Union[int, float]], optional): The minimum
+            frequency for an entity. Any entities with a frequency below this
+            will be filtered. Defaults to None.
+        max_entity_freq (Optional[Union[int, float]], optional): The maximum
+            frequency for an entity. Any entities with a frequency above this
+            will be filtered. Defaults to None.
+
+    Returns:
+        Mapping[str, Mapping[str, Union[str, str]]]: Filtered entities.
+    """
+    entity_freqs = Counter(chain(*entities.values()))
+    entity_freq_low = min(entity_freqs.values())
+    entity_freq_high = max(entity_freqs.values())
+
+    if type(min_entity_freq) == float:
```

Comment:
I am not sure I understand how this works when it `min_entity_freq` is a float.

Reply:
The idea is that you can specify some percentage of the entity with the highest frequency as the minimum threshold frequency.
a) This isn't obvious and I should have specified in the docstring 😅 
b) Using a percentile of the distribution of frequencies as a threshold parameter probably makes more sense on reflection